balancer/weightedtarget/weightedaggregator/aggregator.go

```diff
@@ -128,9 +135,14 @@ func (wbsa *Aggregator) Remove(id string) {
 	if _, ok := wbsa.idToPickerState[id]; !ok {
 		return
 	}
+	// Record transition for picker to connectivity.Shutdown to decrement counter
+	// for the picker's previous state
+	wbsa.csEvltr.RecordTransition(wbsa.idToPickerState[id].stateToAggregate, connectivity.Shutdown)
 	// Remove id and picker from picker map. This also results in future updates
 	// for this ID to be ignored.
 	delete(wbsa.idToPickerState, id)
+	// Call buildAndUpdateLocked to update picker state
```

Comment:
Do you mean why is there a buildAndUpdate-Locked? 

Since `Add()` and `Remove()` already holds lock on the Mutex, we need to call BuildAndUpdateLocked()